rosbag2_py/src/rosbag2_py/format_bag_metadata.cpp

```diff
@@ -124,17 +126,105 @@ void format_topics_with_type(
       info_stream << std::endl;
     };
 
-  print_topic_info(topics[0]);
   size_t number_of_topics = topics.size();
-  for (size_t j = 1; j < number_of_topics; ++j) {
+  size_t i = 0;
+  // Find first topic which isn't service event topic
+  while (i < number_of_topics &&
+    rosbag2_cpp::is_service_event_topic(
+      topics[i].topic_metadata.name,
+      topics[i].topic_metadata.type))
+  {
+    i++;
+  }
+
+  if (i == number_of_topics) {
+    return;
```

Comment:
@MichaelOrlov 

> Maybe output to the info_stream that no topics found? To inform users.

Need not inform user.  
While the recorded file only includes service info, this function is still called.   

> It could be confusing for users when only service events in the recorded bag and ros2 bag info output nothing.

If the recorded file only includes service events, it will be showed as below. What do you think will confuse user?
Do you mean there is nothing following `Topic information: ` ?
```
Files:             abc_0.mcap
Bag size:          6.9 KiB
Storage id:        mcap
ROS Distro:        rolling
Duration:          5.500s
Start:             Oct 16 2023 09:33:20.374 (1697420000.374)
End:               Oct 16 2023 09:33:25.875 (1697420005.875)
Messages:          0
Topic information: 
Service:           1
Service information: Service: /add_two_ints | Type: example_interfaces/srv/AddTwoInts | Event Count: 24 | Serialization Format: cdr
```

Reply:
@Barry-Xu-2018 Disregard my request.
I was under the impression that we would print nothing after the 
```
Messages:          0
Topic information: 
```
Sorry about that.